src/components/Search/SearchList/index.tsx

```diff
@@ -177,15 +178,46 @@ function SearchList({
         }
         return data;
     }, [data, groupBy]);
-    const flattenedItemsWithoutPendingDelete = useMemo(() => flattenedItems.filter((t) => t?.pendingAction !== CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE), [flattenedItems]);
-
-    const selectedItemsLength = useMemo(
-        () =>
-            flattenedItems.reduce((acc, item) => {
-                return item?.isSelected ? acc + 1 : acc;
-            }, 0),
-        [flattenedItems],
-    );
+
+    const emptyReports = useMemo(() => {
+        if (groupBy && isTransactionGroupListItemArray(data)) {
+            return data.filter((item) => item.transactions.length === 0);
+        }
+        return [];
+    }, [data, groupBy]);
+
+    const selectedItemsLength = useMemo(() => {
+        if (groupBy && isTransactionGroupListItemArray(data)) {
+            const selectedEmptyReports = emptyReports.reduce((acc, item) => {
+                return acc + (item.isSelected ? 1 : 0);
+            }, 0);
+
+            const selectedTransactions = flattenedItems.reduce((acc, item) => {
+                return acc + (item?.isSelected ? 1 : 0);
+            }, 0);
+
+            return selectedEmptyReports + selectedTransactions;
+        }
+
+        return flattenedItems.reduce((acc, item) => {
+            return acc + (item?.isSelected ? 1 : 0);
+        }, 0);
+    }, [data, flattenedItems, groupBy, emptyReports]);
```

Comment:
I think we simplify to the following, wdyt?
```suggestion
    const selectedItemsLength = useMemo(() => {
        const selectedTransactions = flattenedItems.reduce((acc, item) => acc + (item?.isSelected ? 1 : 0), 0);
        if (groupBy) {
            const selectedEmptyReports = emptyReports.reduce((acc, item) => acc + (item.isSelected ? 1 : 0), 0);
            return selectedEmptyReports + selectedTransactions;
        }
        return selectedTransactions;
    }, [flattenedItems, groupBy, emptyReports]);
```